src/modals/storage-connections/StorageConnectionsModal.js

```diff
@@ -0,0 +1,154 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import PluginApiModal from '_/components/modals/PluginApiModal'
+import getPluginApi from '_/plugin-api'
+import { msg } from '_/intl-messages'
+import { Spinner } from 'patternfly-react'
+import { Button } from '@patternfly/react-core'
+import StorageConnectionsModalBody from './StorageConnectionsModalBody'
+import { createErrorMessage } from '_/utils/error-message'
+import './connections.css'
+
+const StorageConnectionsModal = ({
+  isLoading = false,
+  storageDomain,
+  domainConnections,
+  allConnections,
+  onClose = () => { },
+  doConnectionCreate = () => { },
+  doConnectionEdit = () => { },
+  doConnectionDelete = () => { },
+  doConnectionAttach = () => { },
+  doConnectionDetach = () => { },
+  doRefreshConnections = () => { },
+}) => {
+  const [isOpen, setOpen] = useState(true)
+  const [error, setError] = useState(null)
+  const [isShowAll, setShowAll] = useState(false)
+
+  const warning = (!storageDomain || storageDomain.status === 'maintenance')
+    ? null
+    : {
+      title: msg.storageConnectionsDomainNotInMaintenanceWarning(),
+      detail: msg.storageConnectionsDomainNotInMaintenanceWarningDetail(),
+    }
+
+  const close = () => {
+    setOpen(false)
+    onClose()
+  }
+
+  const modalActions = [
+    <Button
+      key='storage-connections-modal-cancel-button'
+      variant='primary'
+      onClick={close}
+    >
+      {msg.closeButton()}
+    </Button>,
+  ]
+
+  const createTitle = () => {
+    let title = msg.storageConnectionsTitle()
+    if (storageDomain) {
+      title = title + ' - ' + storageDomain.name
+    }
+    return title
+  }
+
+  const onConnectionCreate = async (connection) => {
+    try {
+      await doConnectionCreate(connection)
+      setShowAll(true)
+      doRefreshConnections()
```

Comment:
since `doRefreshConnections` is always called it seems a part of data provider logic

Reply:
@rszwajko but since it's a callback passed by DataProvider component - I don't really have access to it in StorageConnectionsDataProvider, so the only way to call it inside doConnectionCreate is to pass it as an argument there. Thought I might as well call it here, also I think it is modal logic - I perform an action and then refresh the data. 
Or am I missing something here?

Reply:
>  it's a callback passed by DataProvider component -
>

Yes and you can define the callback there, while creating StorageConnectionsDataProvider i.e. somwhere before
```
 return React.cloneElement(child, {
        storageDomain,
        domainConnections,
        allConnections,
        doConnectionCreate: createConnection,
       .....
```

> I think it is modal logic
>

Ultimately everything what's on the modal is modal logic:) However by using data provider we extract that part to the data provider.

Btw `setShowAll(true)` does not belong here as it should be tied with React lifecycle. By the time the first call will succeed the component may be already unmounted which results in errors.

Reply:
Maybe if it's okay then I would leave it the way it is. In that case, what would I need to sort out? Ignore fetchInProgress to not display the spinner? If yes - should I ignore it during the initial load as well? I tried different fetch errors, and it looked okay, but it is possible that I am missing something

Reply:
Spinner component looks ok.

The current code in `StorageConnectionsDataProvider` will activate the spinner every time a fetch is in progress.  You would need to tweak the data provider to change that behavior.